AFMReader/jpk.py

```diff
@@ -35,7 +35,62 @@ def _jpk_pixel_to_nm_scaling(tiff_page: tifffile.tifffile.TiffPage) -> float:
     return px_to_nm * 1e9
 
 
-# pylint: disable=too-many-locals
+def _get_z_scaling(tif: tifffile.tifffile, channel_idx: int) -> tuple[float, float]:
+    """
+    Extract the z scaling factor and offset for a JPK image channel.
+
+    Determined using JPKImageSpec.txt Version: 2.0:3fffffffffff supplied with JPK instrument.
+
+    Parameters
+    ----------
+    tif : tifffile.tifffile
+        A tiff file of .jpk images.
+
+    Returns
+    -------
+    tuple
+        A tuple contains values used to scale and offset raw data.
+    """
+    NrOfSlots = tif.pages[channel_idx].tags["32896"].value
+    default_slot = tif.pages[channel_idx].tags["32897"]
+
+    # Create a dictionary of list for the differnt slots
+    slots = {}
+    for slot in range(NrOfSlots):
+        slots[slot + 1] = []
+
+    # Extract the tags with numerical names in each slot
+    while NrOfSlots >= 1:
+        for tag in tif.pages[channel_idx].tags:
+            try:
+                tag_name_float = float(tag.name)
+                if tag_name_float >= (32912 + ((NrOfSlots - 1) * 48)) and tag_name_float < (32912 + (NrOfSlots * 48)):
+                    slots[NrOfSlots].append(tag.name)
+            except ValueError:
+                continue
+        NrOfSlots -= 1
+
+    # Find the number of the default slot (selected in the instrument GUI)
+    for slot_name, values in slots.items():
```

Comment:
Here `slot_name` will be an `int` rather than a `str` by virtue of using `n_slot + 1` as keys in the `slots` dictionary, so `_name` could be slightly misleading. I think just `slot` would be fine here (and on line 77).


```suggestion
    for slot, values in slots.items():
```